Providers/Scripts/2.4x-2.5x/Scripts/nxOMSGenerateInventoryMof.py

```diff
@@ -0,0 +1,167 @@
+        instance['ClassName'] = protocol.MI_String(instance['ClassName'])
+        if instance['Properties'] is not None and len(instance['Properties']):
+            instance['Properties'] = protocol.MI_StringA(instance['Properties'])
+    Instances = protocol.MI_InstanceA(CurrentInstances)
+    RunIntervalInSeconds = protocol.MI_Uint64(RunIntervalInSeconds)
+    Tag = protocol.MI_String(Tag)
+    Format = protocol.MI_String(Format)
+    FilterType = protocol.MI_String(FilterType)
+
+    if Configuration is None:
+        Configuration = []
+    if Configuration is not None and len(Configuration):
+        Configuration = protocol.MI_StringA(Configuration)
+
+    retd = {}
+    ld = locals()
+    for k in arg_names:
+        retd[k] = ld[k]
+    return 0, retd
+            
+def Set(FileName, Enable, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration):
+    if Enable == True:
+         GenerateInventoyMOF(FileName, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+    return [0]
+
+def Test(Enable, Instances):
+    return [-1]
+
+def GenerateInventoyMOF(FileName, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration):
+    header = '/*This is an autogenerated file \n@TargetNode=\'Localhost\'*/\n'
+    footer= '\ninstance of OMI_ConfigurationDocument \n{\n \
+    DocumentType = \"inventory\"; \n \
+    Version=\"2.0.0\"; \n \
+    MinimumCompatibleVersion = \"2.0.0\"; \n \
+    Name=\"InventoryConfig\"; \n};'
+
+    inventoryMofSection = ''
+    if Instances is not None:
+        for instance in Instances:
+            instanceName = instance['InstanceName']
+            className = instance['ClassName']
+            filepaths = '    '+';\n    '.join(instance['Properties'])
+            new_source = 'instance of ' + className + '\n{\n'
+            new_source+= filepaths + ';\n};\n'
+            inventoryMofSection+=new_source
+
+    mof_file_path = inventoryMof_path + FileName
+    txt = header + inventoryMofSection + footer 
+
+    mofFileWritten = False
+    if os.path.isfile(mof_file_path): 
+        filechecksum = GetFileChecksum (mof_file_path)
+        stringchecksum = GetStringChecksum(txt)
+        if (filechecksum != stringchecksum):
+           codecs.open(mof_file_path, 'w', 'utf8').write(txt)
+           mofFileWritten = True
+    else:
+        codecs.open(mof_file_path, 'w', 'utf8').write(txt)
+        mofFileWritten = True
+
+
+    fileNameWithoutExtension = os.path.splitext(FileName)[0] 
+    conf_file_path = inventoryMof_path+ fileNameWithoutExtension + '.conf'
+    conf_file_contents = '#This is auto-generated file \n \
+<source> \n \
+  type exec \n \
+  tag ' + Tag + ' \n \
+  command /opt/microsoft/omsconfig/Scripts/PerformInventory.py --InMOF ' + mof_file_path + ' --OutXML /etc/opt/omi/conf/omsconfig/configuration/' + fileNameWithoutExtension + '.xml > /dev/null && cat /etc/opt/omi/conf/omsconfig/configuration/' + fileNameWithoutExtension + '.xml \n \
```

Comment:
make "/etc/opt/omi/conf/omsconfig/configuration/" a variable like the inventoryMof_path